pyannote/audio/models/models.py

```diff
@@ -526,3 +527,129 @@ def intermediate_dimension(self, layer):
         if layer == 0:
             return self.sincnet_.dimension
         return self.rnn_.intermediate_dimension(layer-1)
+
+
+class SincTDNN(Model):
+    """waveform -> SincNet -> XVectorNet (TDNN -> FC) -> output
+
+    Parameters
+    ----------
+    sincnet : `dict`, optional
+        SincNet parameters. Defaults to `pyannote.audio.models.sincnet.SincNet`
+        default parameters.
+    tdnn : `dict`, optional
+        Time-Delay neural network parameters.
+        Defaults to `pyannote.audio.models.tdnn.XVectorNet` default parameters.
+    embedding : `dict`, optional
+        Embedding parameters. Defaults to `Embedding` default parameters. This
+        only has effect when model is used for representation learning.
+    """
+
+    @staticmethod
+    def get_alignment(sincnet=None, **kwargs):
+        """
+        """
+
+        if sincnet is None:
+            sincnet = dict()
+
+        return SincNet.get_alignment(**sincnet)
+
+    supports_packed = False
+
+    @staticmethod
+    def get_resolution(sincnet : Optional[dict] = None, **kwargs) -> Resolution:
+        """Get sliding window used for feature extraction
+
+        Parameters
+        ----------
+        sincnet : dict, optional
+
+        Returns
+        -------
+        sliding_window : `pyannote.core.SlidingWindow` or {`window`, `frame`}
+        """
+
+        if sincnet is None:
+            sincnet = dict()
+
+        return SincNet.get_resolution(**sincnet)
+
+    def init(self,
+             sincnet : Optional[dict] = None,
+             tdnn : Optional[dict] = None,
+             embedding : Optional[dict] = None):
+        """waveform -> SincNet -> RNN [-> merge] [-> time_pool] -> FC -> output
+
+        Parameters
+        ----------
+        sincnet : `dict`, optional
+            SincNet parameters. Defaults to `pyannote.audio.models.sincnet.SincNet`
+            default parameters.
+        tdnn : `dict`, optional
+            Time-Delay neural network parameters.
+            Defaults to `pyannote.audio.models.tdnn.XVectorNet` default parameters.
+        embedding : `dict`, optional
+            Embedding parameters. Defaults to `Embedding` default parameters. This
+            only has effect when model is used for representation learning.
+        """
+
+        n_features = self.n_features
+
+        if sincnet is None:
+            sincnet = dict()
+        self.sincnet = sincnet
+
+        if n_features != 1:
+            raise ValueError('SincNet only supports mono waveforms. '
+                             f'Here, waveform has {n_features} channels.')
+        self.sincnet_ = SincNet(**sincnet)
+        n_features = self.sincnet_.dimension
+
+        if tdnn is None:
+            tdnn = dict()
+        self.tdnn = tdnn
+        self.tdnn_ = XVectorNet(n_features, **tdnn)
```

Comment:
The naming convention is confusing: `self.tdnn_` is an `XVectorNet` instance, not a `TDNN` instance.

Reply:
I'll fix this before merging